control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -286,7 +290,44 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 		return err
 	}
 
-	address, _ := url.Parse(fmt.Sprintf("http://%s.%s.svc.%s", channelService.Name, channelService.Namespace, network.GetClusterDomainName()))
+	var addressableStatus duckv1.AddressStatus
+	channelHttpsHost := network.GetServiceHostname(r.Env.IngressName, r.SystemNamespace)
+	channelHttpHost := network.GetServiceHostname(channelService.Name, channel.Namespace)
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
+		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		// Permissive mode:
+		// - status.address http address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		//   - http address with path-based routing
+		addressableStatus.Addresses = []duckv1.Addressable{httpsAddress, httpAddress}
+		addressableStatus.Address = &httpAddress
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		// Strict mode: (only https addresses)
+		// - status.address https address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		addressableStatus.Addresses = []duckv1.Addressable{httpsAddress}
+		addressableStatus.Address = &httpsAddress
+	} else {
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
+		addressableStatus.Address = &httpAddress
```

Comment:
It's probably worth setting `Status.Addresses` as well in the default branch
```suggestion
		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
		addressableStatus.Address = &httpAddress
		addressableStatus.Addresses = []duckv1.Addressable{httpAddress}
```

Reply:
In the [Disabled Transport Encryption section of the TLS Proposal](https://docs.google.com/document/d/1H-x_oji8LqkCyd7tlsSyclmUe7FAmEJPgRxOU_0pkn8/edit?resourcekey=0-lzDIPJsZOP3G17QE_g1lHw#heading=h.kzw081mf0knb), It says,

> Addressable resources will keep setting their status.address.url to whatever value their are currently setting, which in most cases is an HTTP cluster-internal endpoint.

> Event producers will keep using status.address.url endpoint as sink address. 

Since we will not be using `Addresses`, I thought it would be better if I don't add addresses to avoid confusion, 

@Cali0707 WDYT?

Reply:
@Vishal-Chdhry I personally think it might be better to just set it anyways, so that we can stay consistent. Currently broker and sink always set `Status.Addresses`. @pierDipi what do you think?

Reply:
Both options seems ok, I think I like that we keep consistency, I think we're setting them in Eventing core https://github.com/knative/eventing/commit/24fbfe500ce6536afdaa0dfa2604eafa4dfbcbb0, so I'd set `addresses` as well.